### ISSUE TITLE
[HOLD for payment 2024-12-03] Add a step to to Request Physical Card form that collects a magic code

### ISSUE THREAD

**Issue by NikkiWines** [Expensify/App#50967]
# Problem
Someone can issue a physical or virtual Expensify card without verifying they are the owner of the account. This relates to an [internal security issue](https://github.com/Expensify/Expensify/issues/434451).

## Why this is important to solve
This is a security vulnerability that can be taken advantage of if an account is compromised.

# Solution
Collect a magic code when requesting a physical Expensify card. In a little more detail:

- The API command to be updated is [here](https://github.com/Expensify/App/blob/286212cc4f23b94426dca0bc28a039f61ca5c693/src/libs/actions/Wallet.ts#L260-L294)
- It should have a new parameter called `validateCode` that is passed to the server
- In order to do this, anything callling `BaseGetPhysicalCard` needs a new step to gather a magic code from the user (we should have existing components that can be reused for this)

<details><summary>Issue Owner</summary>Current Issue Owner: @sakluger</details>

**Comment by NikkiWines** [comment]
cc: @mountiny, sounds like from [this](https://github.com/Expensify/App/issues/50667#issuecomment-2416202867) we're still working on the re-usable component for requesting a validateCode, is that correct?

**Comment by hungvu193** [comment]
Not sure why I'm getting notification for this issue but yeah we have an issue for this kind of flow. It's being worked on by @getusha in https://github.com/Expensify/App/pull/49445.

**Comment by sakluger** [comment]
Should we assign @getusha to this issue, or is this issue a duplicate, or do we still need someone else for this one?

**Comment by mountiny** [comment]
I think we can, or @hungvu193 do you want to take on the implementation here?

**Comment by hungvu193** [comment]
Ok, I can take it.

**Comment by mountiny** [comment]
Happy to help as CME

**Comment by hungvu193** [comment]
Draft PR is [here](https://github.com/Expensify/App/pull/51135). It's based on https://github.com/Expensify/App/pull/49445 so I will wait for https://github.com/Expensify/App/pull/49445 to be merged.

**Comment by mountiny** [comment]
Not overdue

**Comment by mountiny** [comment]
The PR was merged, can you sync up with main and make the PR ready for a review @hungvu193? Thank you!

**Comment by hungvu193** [comment]
Sure thing. Also @NikkiWines I noticed that currently we can pass any validate code and BE still returns success response. I think we still need to update BE right?

**Comment by NikkiWines** [comment]
Yep, there's a backend component in the works, it's the last thing we'll update though so as to not break any front-end flows that still need changing

**Comment by hungvu193** [comment]
Cool. I merged FE's PR with main, it's basically ready, only handling error issues left.

**Comment by situchan** [comment]
Not overdue

**Comment by situchan** [comment]
Not overdue

**Comment by hungvu193** [comment]
Please let me know when BE is ready so I can update the FE's PR @NikkiWines

**Comment by NikkiWines** [comment]
Will do! We're waiting on another PR but it's in the works

**Comment by situchan** [comment]
Please reassign another C+ as I'll be OOO

**Comment by sakluger** [comment]
@dominictb I sent you an offer through Upwork: https://www.upwork.com/nx/wm/offer/105165129

**Comment by sakluger** [comment]
@hungvu193 please request payment using the payment summary comment above.

**Comment by garrettmknight** [comment]
$250 approved for @hungvu193
